docs/reference-guides/rancher-manager-architecture/communicating-with-downstream-user-clusters.md

```diff
@@ -76,6 +76,85 @@ With this endpoint enabled for the downstream cluster, Rancher generates an extr
 
 You will need to use a context defined in this kubeconfig file to access the cluster if Rancher goes down. Therefore, we recommend exporting the kubeconfig file so that if Rancher goes down, you can still use the credentials in the file to access your cluster. For more information, refer to the section on accessing your cluster with [kubectl and the kubeconfig file.](../../how-to-guides/new-user-guides/manage-clusters/access-clusters/use-kubectl-and-kubeconfig.md)
 
+## Impersonation
+
+Users technically exist only on the upstream cluster. When users interact with a downstream cluster through the authentication proxy, there needs to be some entity downstream to serve as the actor for those requests. Rancher creates service accounts to be that entity. Each service account is only granted one permission, which is to **impersonate** the user they belong to. If there was only one service account that could impersonate any user, then it would be possible for a malicious user to corrupt that account and escalate their privileges by impersonating another user. This issue was the basis for a [CVE](https://github.com/rancher/rancher/security/advisories/GHSA-pvxj-25m6-7vqr).
```

Comment:
We should add some wording to specify how we determine downstream permissions. At first, I was confused about how the service account can perform action if it only has impersonation rules.
```suggestion
Users technically exist only on the upstream cluster. Eventhough all RoleBindings and ClusterRoleBindings on the downstream refer to the userInfo stored in the local cluster. When users interact with a downstream cluster through the authentication proxy, there needs to be some entity downstream to serve as the actor for those requests. Rancher creates service accounts to be that entity. Each service account is only granted one permission, which is to **impersonate** the user they belong to. If there was only one service account that could impersonate any user, then it would be possible for a malicious user to corrupt that account and escalate their privileges by impersonating another user. This issue was the basis for a [CVE](https://github.com/rancher/rancher/security/advisories/GHSA-pvxj-25m6-7vqr).
```

Reply:
I'm not sure I understand. Doesn't the CR/CRBs in downstream refer to the `user-*` created in the downstream?

Reply:
Alright, less confused now.

> Eventhough all RoleBindings and ClusterRoleBindings on the downstream refer to the userInfo stored in the local cluster. 

I don't think userInfo is mentioned anywhere in the Rancher docs so that might confuse people that don't know about it.

Reply:
Added some references to upstream RBAC docs for clarity